### DISCUSSION TITLE
Windows 10 Build Tips

### DISCUSSION THREAD

**Discussion by ddesmond** [AcademySoftwareFoundation/OpenRV#91] [Build Issues] [answered]
For those on Windows 10 platform here are some tips to make rv.exe compile.
As of March 1st/2nd there are couple things that needs to be tweaked to get it "all "working.

Downloading MSYS from https://www.msys2.org/ will get you a version which needs to be updated with the following commands.
Run MSYS MinGW64 and do:
```
pacman -S cmake
pacman -S python
pacman -S pip
pacman -S python-pip
```
Provided cmake version is 3.20, this will update to 3.25

You need to setup ONE aclocal to the aclocal PATH, do this (in same console we did the update to)
`export ACLOCAL_PATH=/usr/share/aclocal`

Switch to /c/OpenRV and open the rvcmds.sh to edit it
This is needed otherwise the python deps will fail with subprocess exit errors
change:
`alias rvsetup="python3 -m pip install --user --upgrade -r ${RV_HOME}/requirements.txt"`
to
`alias rvsetup="SETUPTOOLS_USE_DISTUTILS=stdlib pip install --user --upgrade -r ${RV_HOME}/requirements.txt"`

This will get rid of the python pip errors and make it work and install.

you can then 
`source rvcmds.sh` and boostrap will work.

then you can try:
`cmake -B _build -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64`
respective to the perl and QT paths you have installed to

and:
`cmake --build _build --config Release -v --parallel=12 --target main_executable > buildlog.log`
to have a silent compile and all sent to log file buildlog
or 
`cmake --build _build --config Release -v --parallel=8 --target main_executable`

The `-parallel `arg is how many parallel jobs you can run, depending on your cpu count you can set it higher or lower (more cpu count more faster compile)

You now have a build in _builds/stage/app in bin directory there is rv.exe (or at least it should be) and you can try running it

The step to install is:
`cmake --install _build --prefix _install`

This will create _install dir in the openrv root dir and you can try running it from there.

Stripping debug stuf for me doesnt work at all, and I end up with no rv.exe if i provide anything other than that command.

It would be great if maintaniers would specify certain stuff and fix the build issues or provide a bat or sh file for windows platform.

Not all users are TDs or developers, and this would help imensely to anyone who wants to run OpenRV no his machine (as many do it seems)

Anyway, thanks to all devs to providing this, with a bit of pain its workable.

Missing $HOME
If you start it from network or in certain scenarios, on run openrv.exe can show $HOME varable missing, setup your $HOME variable to your user home folder somewhere and the app will run again.

If you happen to  install pacakges and the app just closes to desktop, simplest fix is to keep an _install or build app folder copied so you can restore it (packages copy files around to rv, like ftrack-connect-rv, which will botch your running app and you wont be able to run it again)

Edit: Typos, and missing home-install packages

**Comment by BrianHanke** [comment]
Great tips, thanks! The aclocal one in particular seems to be a common problem and impossible to know how to fix without specialized knowledge. Would be good to have a note about it in the official docs.

**Comment by geffrak** [comment]
Improving the windows build experience is one of our current priorities. You should see improvements on that front soon.

---

The full build is aliased to `rvbootstrap` (or `rvbootstrapd` for debugging). It excludes installation. 
The install step is aliased to `rvinst`.

This is why we don't provide a bat/sh file. It's the same command for all operating systems and should work well once you get used to it. You can `rvbootstrap && rvinst` to have it fully automated. 

---

We could add the `ACLOCAL_PATH` environment variable to https://github.com/AcademySoftwareFoundation/OpenRV/blob/2b799c585d031c022f241bc02e454b26654c3a61/rvcmds.sh#L29-L36 (Pull request welcomed) 

---

`SETUPTOOLS_USE_DISTUTILS` is set since my last pull request https://github.com/AcademySoftwareFoundation/OpenRV/blob/2b799c585d031c022f241bc02e454b26654c3a61/rvcmds.sh#L35 so you should not have to edit the rvcmds.sh

---

Redirecting to a build log is a good idea. You can redirect any alias to a file. Perhaps we could do it automatically using `tee`.

**Reply by ddesmond** [reply]
SETUPTOOLS_USE_DISTUTILS is set since my last pull request

somehow its not in the main, altough i see it as merged.

**Reply by rubikstriangle** [reply]
I recently noticed that there are some handy pics that make parts of the windows setup clear.

https://github.com/AcademySoftwareFoundation/OpenRV/blob/main/docs/build_system/win_microsoft_visual_studio_features_to_install.png
https://github.com/AcademySoftwareFoundation/OpenRV/blob/main/docs/build_system/win_how_to_enable_dot_NET_3_5.png

Might be good to embed these images into the docs so that they are not overlooked.

**Reply by geffrak** [reply]
@ddesmond Are you working from a fork? If yes, you need to sync your fork.

@rubikstriangle They were, they got removed by mistake.

**Reply by ddesmond** [reply]
> m a fork? If yes, you need to sy

nope, i pulled fresh on the 1st march. I do see SETUPTOOLS but it seems its not being evaluated properly on my end or those lines are just being ignored.

**Comment by ahemberger** [comment]
Hi there, I'm trying to muddle my way through a windows build, I'm finding following the docs confusing, and found this thread which indeed is helpful, so thank you so much for contributing! To set the tone, I'm a TD who used RV for years in studio, but this build process is well beyond my developer skillz.

A problem I'm running into: I use Anaconda to manage Python on my windows machine. I'm not an expert with it, but can successfully create various python environments to use for things like my own development or testing our various github projects like stable_diffusion and whatnot. In each case, I seem to have to launch a shell using that particular environment in order to use that particular python version. 

I'm getting confused with what seems to be the use of single-purpose shells that don't seem to cross-talk; I can open a Git bash shell and use unix-like commands, but can't run these rv scripts in it. The rvbootstrap or rvcmds.sh scripts also don't seem to be useful in Windows terminals, powershell, etc. This thread put me onto using MinGW64 (I have no idea what this is; looks like another new shell, with apparently a new set of available commands), which does allow me to get a bit further, but eventually the rvbootstrap process fails, with errors that appear to be related to python. I'm not all the way surprised, but my question is: is there a way to get these two things (one of anaconda's python environments and whatever MinGW64 is) to play together in order to muddle further through this build?

Thank you!

**Reply by rubikstriangle** [reply]
I just did a fresh install on windows 10 and didn't have to do anything special with python.  Where is it breaking down? Some errors might be helpful.

**Reply by ahemberger** [reply]
Sure:

The first thing that's going differently than what's described above is when I step through these commands:
```
pacman -S cmake
pacman -S python
pacman -S pip
pacman -S python-pip

```
the pip step fails:
```
$ pacman -S pip
error: target not found: pip
```

Now, I mentioned I'm a bit of a n00b, so it's not obvious to me what these steps are doing; I assume they're connecting what's meant to be a system python install to whatever this MINGW64 thing is, but maybe they're actually installing totally separate instances of python/cmake and such? In any case, it's not clear to me why the pip step fails or how I can fix that. The others (including the following python-pip step) seem to work. 

I'll note that the python that is mentioned in the output from running the above python step is 3.11. This might be relevant a bit later here. 

Anyway, I soldier on blissfully unaware where I am already doomed to failure. My next confusing point is ` source rvcmds.sh`, where the default paths reported for things seem to use a confusing mishmash of filepath conventions:

```
$ source rvcmds.sh
Please ensure you have installed any required dependencies from doc/build_system/config_[os]


CMake parameters:
RV_HOME is /h/github/OpenRV
RV_BUILD is /h/github/OpenRV/_build
RV_INST is /h/github/OpenRV/_install
CMAKE_GENERATOR is Visual Studio 16 2019
QT_HOME is c:/Qt/Qt/5.15.2/msvc2019_64
WIN_PERL is c:/Strawberry/perl/bin
To override any of them do unset [name]; export [name]=value; source -bash
```
The defaults for strawberry perl and qt don't match what I did (my machine has multiple drives, these two things live on h:/ for me, or is it /h/?) I choose to roll the dice and make the paths all look more unix-y, so I change things to:

```
CMake parameters:
RV_HOME is /h/github/OpenRV
RV_BUILD is /h/github/OpenRV/_build
RV_INST is /h/github/OpenRV/_install
CMAKE_GENERATOR is Visual Studio 16 2019
QT_HOME is /h/github/qt/5.15.2/msvc2019_64
WIN_PERL is /h/github/strawberry_perl/perl/bin
```
The advice above next says once I've done this "rvbootstrap will work", which suggests to me that's the next step I should undertake (again, these docs aren't totally clear to someone who's never done this before). rvbootstrap seems to do a lot of things...I see what looks like some python stuff being installed, (into my c:/user/.local/lib/python3.10 site-packages area). Again, the `pacman -S python` step above mentioned 3.11...so it there a disconnect here? 

Next I see a patch of statements complaining about msys64 flex.exe and sed.exe not being found, but the text isn't red, so maybe ok? The script keeps going...

Then an absolute ton of output, with a handful of yellow and red errors, unsure whether I should expect these or not. 

Then finally, BUILD FAILED, and this:
```

Build FAILED.

       "H:\github\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "H:\github\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "H:\github\OpenRV\_build\compiled_python_source_modules.vcxproj" (default target) (7) ->
       "H:\github\OpenRV\_build\src\bin\python\py-interp\pyc_generation.vcxproj" (default target) (11) ->
       "H:\github\OpenRV\_build\installed_packages.vcxproj" (default target) (13) ->
       "H:\github\OpenRV\_build\src\bin\apps\rvpkg\rvinstall.vcxproj" (default target) (18) ->
       "H:\github\OpenRV\_build\packages.vcxproj" (default target) (21) ->
       "H:\github\OpenRV\_build\src\plugins\rv-packages\annotate\annotate-1.15.rvpkg.vcxproj" (default target) (127) ->
       (CustomBuild target) ->
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): warning MSB8064: Custom build for item "H:\github\OpenRV\_build\CMakeFil
       es\7516c3839a43631f2c994cfe64419f04\annotate-1.15.rvpkg.rule" succeeded, but specified dependency "h:\github\openrv\src\plugins\rv-packages\annotate\source_icons" does not exist. This may cause incr
       emental build to work incorrectly. [H:\github\OpenRV\_build\src\plugins\rv-packages\annotate\annotate-1.15.rvpkg.vcxproj]


       "H:\github\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "H:\github\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "H:\github\OpenRV\_build\cmake\dependencies\dependencies.vcxproj" (default target) (17) ->
       "H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default target) (57) ->
         CUSTOMBUILD : warning : The script f2py.exe is installed in 'H:\github\OpenRV\_build\RV_DEPS_PYTHON3\install\Scripts' which is not on PATH. [H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTH
       ON3-stage-target.vcxproj]
         CUSTOMBUILD : CMake warning :  [H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         H:\github\OpenRV\_build\_deps\rv_deps_pyside2-src\sources\shiboken2\libshiboken\sbkenum.cpp(510): warning C4138: '*/' found outside of comment [H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_P
       YTHON3-stage-target.vcxproj]


       "H:\github\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "H:\github\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "H:\github\OpenRV\_build\executables_with_plugins.vcxproj" (default target) (14) ->
       "H:\github\OpenRV\_build\src\bin\imgtools\makeFBIOformats\makeFBIOformats.vcxproj" (default target) (20) ->
       "H:\github\OpenRV\_build\mu_plugins.vcxproj" (default target) (59) ->
       "H:\github\OpenRV\_build\src\lib\mu\mu_encoding\mu_encoding.vcxproj" (default target) (62) ->
       "H:\github\OpenRV\_build\src\lib\mu\MuEncoding\MuEncoding.vcxproj" (default target) (87) ->
       "H:\github\OpenRV\_build\src\lib\mu\MuLang\MuLang.vcxproj" (default target) (91) ->
       (ClCompile target) ->
         H:\github\OpenRV\src\lib\mu\MuLang\MuLang/FlexLexer.h(55,19): fatal error C1017: invalid integer constant expression (compiling source file H:\github\OpenRV\src\lib\mu\MuLang\MuLang\generated\Flex
       Lexer.cpp) [H:\github\OpenRV\_build\src\lib\mu\MuLang\MuLang.vcxproj]
         H:\github\OpenRV\src\lib\mu\MuLang\MuLang/FlexLexer.h(55,19): fatal error C1017: invalid integer constant expression (compiling source file H:\github\OpenRV\src\lib\mu\MuLang\MuLang\generated\Gram
       mar.cpp) [H:\github\OpenRV\_build\src\lib\mu\MuLang\MuLang.vcxproj]


       "H:\github\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "H:\github\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_BOOST.vcxproj" (default target) (10) ->
       (CustomBuild target) ->
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'H:\github\OpenRV\_build\CMakeFiles\f14f
       2dfb8c8715bfad812616e8d16453\RV_DEPS_BOOST-configure.rule;H:\github\OpenRV\_build\CMakeFiles\f14f2dfb8c8715bfad812616e8d16453\RV_DEPS_BOOST-build.rule;H:\github\OpenRV\_build\CMakeFiles\f14f2dfb8c87
       15bfad812616e8d16453\RV_DEPS_BOOST-install.rule;H:\github\OpenRV\_build\CMakeFiles\7ecdf1b7839ce57c1f936665c71ea32b\RV_DEPS_BOOST-complete.rule;H:\github\OpenRV\_build\CMakeFiles\a1ab24d6bbbb83f2552
       360de8c5a3de2\RV_DEPS_BOOST.rule' exited with code 1. [H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_BOOST.vcxproj]


       "H:\github\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "H:\github\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "H:\github\OpenRV\_build\cmake\dependencies\dependencies.vcxproj" (default target) (17) ->
       "H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default target) (57) ->
         CUSTOMBUILD : error : pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts. [H:\gi
       thub\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         H:\github\OpenRV\_build\_deps\rv_deps_pyside2-src\sources\shiboken2\libshiboken\sbknumpyarrayconverter.cpp(119): error C2065: 'NPY_ARRAY_UPDATEIFCOPY': undeclared identifier [H:\github\OpenRV\_bui
       ld\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         CUSTOMBUILD : error : Error compiling shiboken2 [H:\github\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'H:\github\OpenRV\_build\CMakeFiles\4dc7
       6ff5c3feb07293c123abff2fa893\RV_DEPS_PYTHON3-requirements-flag.rule;H:\github\OpenRV\_build\CMakeFiles\4dc76ff5c3feb07293c123abff2fa893\RV_DEPS_PYSIDE2-build-flag.rule;H:\github\OpenRV\_build\CMakeF
       iles\c2db6fc802aac52b66d634ec7fdbe50a\python39.dll.rule;H:\github\OpenRV\_build\CMakeFiles\a1ab24d6bbbb83f2552360de8c5a3de2\RV_DEPS_PYTHON3-stage-target.rule' exited with code 1. [H:\github\OpenRV\_
       build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

    4 Warning(s)
    7 Error(s)

Time Elapsed 00:02:28.75
```
I don't know which bits are important above; there is, for example, an error about an expectation of a path like h/github/openrv, whereas my actual folder is h/github/OpenRV, eg.

**Reply by rubikstriangle** [reply]
Nothing jumps out at me, I'm perhaps not the most qualified person at diagnosing these logs though.  My first few builds failed, then I created a fresh install of windows and installed all of the programs in default locations in a verbatim following of the guides (the one above).  Then I was able to get it to work.  My most recent build was on a windows 10 virtual machine.  I tried following the steps in the above guide but had trouble getting all of the pip installs to work, eventually I got them (If I remember correctly by removing --user from the command).  I was never able to get "pacman -S pip" to work, same error you reported,  but the RV build still worked for me.  When the build finished It seems like I had over 7000 warnings with no errors.

**Reply by tokejepsen** [reply]
@ahemberger have you got Windows WSL disabled?
This helped me with the FlexLexer errors.

**Reply by dkeraudren** [reply, marked as answer]
> @ahemberger have you got Windows WSL disabled? This helped me with the FlexLexer errors.

You mean you need to disable wsl or enbale it to make it work ? I  got the same error and wsl is surely enabled

**Reply by tokejepsen** [reply]
> You mean you need to disable wsl or enbale it to make it work ? I got the same error and wsl is surely enabled

Disabled.

**Reply by dkeraudren** [reply]
> > You mean you need to disable wsl or enbale it to make it work ? I got the same error and wsl is surely enabled
> 
> Disabled.

Unfortunatelly, removing wsl didn't do the trick for me. Building open rv is such a pain omg

**Reply by ahemberger** [reply]
yeah, unfortunately I gave up after getting this advice - the build process felt inhospitable and very difficult, and having to noodle around with system configurations to get an image viewer going isn't appealing to someone who's already invested years of effort getting the rest of a VFX pipeline up and running at home...it's already tenuous and requires layers of understanding, balancing various build environments, python installations, OCIO configs, etc. etc.

**Comment by AgaveCactus** [comment]
Thank you @ddesmond for the super helpful guide. I followed it to the T and still getting build error. Looks like it might be OpenSSL related but I'm far from being an expert at this.  Does anybody have an idea on what went wrong?
[error.txt](https://github.com/AcademySoftwareFoundation/OpenRV/files/11651753/error.txt)

**Comment by A6i8** [comment]
Thak you @ddesmond for this helpful guide but i am getting build error. Does anybody have an idea?

```
"C:\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "C:\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3.vcxproj" (default target) (10) ->
       "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj" (default target) (21) ->
       (CustomBuild target) ->
         NMAKE : fatal error U1077: 'C:\msys64\usr\bin\nasm.EXE' : return code '0x3' [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
         NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return c
       ode '0x2' [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for
        'C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-mkdir.rule;C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEP
       S_OPENSSL-download.rule;C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-update.rule;C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961
       ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-patch.rule;C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-configure.rule;C:\OpenRV\_build\CMak
       eFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-build.rule;C:\OpenRV\_build\CMakeFiles\cc66a7a77d5b0961ebdb690bfdfa9eb2\RV_DEPS_OPENSSL-install.rule;
       C:\OpenRV\_build\CMakeFiles\ddfe0fbca828d26d63401d1d79fd8a3b\RV_DEPS_OPENSSL-complete.rule;C:\OpenRV\_build\CMakeFiles\dc0b82080efe119cccb8b58ef9885864\RV_DE
       PS_OPENSSL.rule;C:\OpenRV\cmake\dependencies\CMakeLists.txt' exited with code 1. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
```

**Reply by naotakahoriguchi** [reply]
same as me! @A6i8 
https://github.com/AcademySoftwareFoundation/OpenRV/discussions/175#discussioncomment-6248033

**Reply by jcuthbertCAS** [reply]
I'm also getting the exact same build error. Any suggestions would be greatly appreciated.

**Reply by geffrak** [reply]
https://github.com/AcademySoftwareFoundation/OpenRV/pull/197

**Reply by scottmcdonnell** [reply]
I got this when my path for WIN_PERL had a typo. it needs to be `<install location>/Strawberry/perl/bin`
check the path is correctly set in `source rvcmds.sh` then run a `rvclean` before `rvbootstrap`

**Comment by aforanup** [comment]
cannot figure out this error... help

  1>Checking Build System
  1>Creating directories for 'rv_deps_pyside2-populate'
  Performing download step (git clone) for 'rv_deps_pyside2-populate'
  Cloning into 'rv_deps_pyside2-src'...
  HEAD is now at 9282e03de Enable 5.15.2.1 fix release
  Submodule 'sources/pyside2-tools' (git://code.qt.io/pyside/pyside-tools.git) registered for path 'sources/pyside2-tools'
  Cloning into 'C:/OpenRV/_build/_deps/rv_deps_pyside2-src/sources/pyside2-tools'...
  Submodule path 'sources/pyside2-tools': checked out 'a8448837204faee0b457d1e2d4cbf574a2811114'
  Building Custom Rule C:/OpenRV/_build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt
  Performing update step for 'rv_deps_pyside2-populate'
  No patch step for 'rv_deps_pyside2-populate'
  No configure step for 'rv_deps_pyside2-populate'
  No build step for 'rv_deps_pyside2-populate'
  No install step for 'rv_deps_pyside2-populate'
  No test step for 'rv_deps_pyside2-populate'
  Completed 'rv_deps_pyside2-populate'
  Building Custom Rule C:/OpenRV/_build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt
-- Using Boost:        1.76.0
-- Using Dav1d:        1.0.0
-- Using FFMPEG:       n4.4.3
-- Using atomic_ops:   7.7.0
-- Using Imath:        3.1.5
-- Using OpenEXR:      3.1.7
-- Using Python3:      3.9.15
-- Using PySide2:      5.15.2.1
-- Using Qt5:          5.15.2
-- Using zlib:         1.2.13
CMake Error at cmake/macros/rv_stage.cmake:422 (EXECUTE_PROCESS):
  EXECUTE_PROCESS error getting child return code: The system cannot find the
  file specified
Call Stack (most recent call first):
  src/plugins/rv-packages/additional_nodes/CMakeLists.txt:11 (RV_STAGE)


-- Configuring incomplete, errors occurred!

**Reply by jcuthbertCAS** [reply]
I'm not sure if this was the exact same error I got, but as long as "rvsetup" runs okay, I next ran "rvcfg -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64" and it completed with no errors.
Afterwards I was able to run rvbuild and rvinst with no issues.

**Reply by BrandonGTS** [reply]
having the same issue here....;w;

**Reply by bernie-laberge** [reply]
Hello @BrandonGTS,
My guess is that the rvcmds.sh script did not set the QT_HOME env var correctly to match your Qt installation.
I always need to set the QT_HOME environment variable explicitly in my `C:\msys64\home\[your_user_name]\.bashrc `to work around this issue:
`export QT_HOME=c:/Qt/5.15.2/msvc2019_64 `

That's because there is no intelligence whatsoever to detect the Open RV builder's Qt installation path in rvcmds.sh.
I'll propose to add some intelligence to it to so that no one faces the same issue again.

**Reply by BrandonGTS** [reply]
OMG!!!! Thank you @bernie-laberge !!! You saved my life!!!! That works!

**Comment by richardssam** [comment]
I've made a branch https://github.com/richardssam/OpenRV/tree/windows-config with some of the changes mentioned here, and a few others.
It would be good to get confirmation that this helps. Thanks again to @ddesmond for making this, I wouldn't have been able to make it without these notes.

The branch is essentially these notes, I did also add a path
 export PATH=$PATH:${RV_HOME}/_build/stage/app/bin

which seemed necessary for some of the building. Also, I ended up defaulting to MSVC 2017 for everything to make the build process easier.

**Comment by ddesmond** [comment]
thank everybody mentioning me but this is a group effort to find out a way to compile it!
also, apology i was burried with work.... but i see you gents and ladies made it work! glad that the notes shed some light.
Again for the repo maintainers, this is not the way how stuff should be pushed to opensource and some random dudes from internet makes notes how to do it.

autodesk you have more devs than i am old :) up up your game (no offense to anyone, its just a positive critique to make stuff better in the OS space)

**Comment by wbpa-garrett** [comment]
Hello,

I hope I'm not too late to this discussion, but I'm having issues building OpenRV on Windows also. I've cleared most of them up but `RV_DEPS_FFMPEG` is still failing during configure due to `openssl not found`. I tried using [this patch ](https://github.com/microsoft/vcpkg/issues/9693#issuecomment-574588505) to fix that one line, but still no luck. Any ideas what I could have missed? The `RV_DEPS_OPENSSL/install` folder looks ok to me, and the rename patch for ffmpeg has been applied. I did pass -DSKIP_CODEC_CHECKS to rvcfg, in case that could cause issues.

Note: I am using vcpkg to install Qt because that's a pain and I'm not normally involved in building on Windows. I've had to restructure the vcpkg dir a bit to build PySide2, but that build works fine so I assume it's not related to this.

Error:
`1>CUSTOMBUILD : error : openssl not found [C:\Users\garrett\wbpa\git\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPEG.vcxproj]`

Contents of the openssl dir:
```
ls git/OpenRV/_build/RV_DEPS_OPENSSL/install/*
RV_DEPS_OPENSSL/install/bin:
c_rehash.pl*            libcrypto-1_1-x64.pdb  libssl-1_1-x64.pdb  openssl.pdb
libcrypto-1_1-x64.dll*  libssl-1_1-x64.dll*    openssl.exe*
RV_DEPS_OPENSSL/install/include:
openssl/
RV_DEPS_OPENSSL/install/lib:
crypto.lib  engines-1_1/  libcrypto.lib  libssl.lib  ssl.lib
```

configure file lines throwing the error:
```
enabled openssl           && { check_pkg_config openssl openssl openssl/ssl.h OPENSSL_init_ssl ||
                               check_pkg_config openssl openssl openssl/ssl.h SSL_library_init ||
                               check_lib openssl openssl/ssl.h OPENSSL_init_ssl -lssl -lcrypto ||
                               check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto ||
                               check_lib openssl openssl/ssl.h SSL_library_init -lssl32 -leay32 ||
                               check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||
                               die "ERROR: openssl not found"; }
```

Line from the vcxproj file:
`C:\msys64\mingw64\bin\cmake.exe -E env PKG_CONFIG_PATH="$"PKG_CONFIG_PATH:/C/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_DAV1D/install/lib/pkgconfig sh ./configure --prefix=C:/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_FFMPEG/install --enable-shared --disable-static --disable-iconv --disable-outdevs --disable-programs --disable-large-tests --disable-vaapi --disable-doc --toolchain=msvc --disable-decoder=bink --disable-decoder=binkaudio_dct --disable-decoder=binkaudio_rdft --disable-decoder=vp9 --disable-decoder=vp9_cuvid --disable-decoder=vp9_mediacodec --disable-decoder=vp9_qsv --disable-decoder=vp9_rkmpp --disable-decoder=vp9_v4l2m2m --disable-decoder=dnxhd --disable-decoder=prores --disable-decoder=qtrle --disable-decoder=aac --disable-decoder=aac_fixed --disable-decoder=aac_latm "--disable-decoder=dvvideo --disable-encoder=dnxhd" --disable-encoder=prores --disable-encoder=qtrle --disable-encoder=aac --disable-encoder=aac_mf --disable-encoder=vp9_qsv --disable-encoder=vp9_vaapi "--disable-encoder=dvvideo --disable-filter=geq --disable-parser=vp9 --disable-protocol=ffrtmpcrypt" --disable-protocol=rtmpe --disable-protocol=rtmpte --extra-cflags=-IC:/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_OPENSSL/install/include --extra-cflags=-IC:/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_DAV1D/install/include --extra-ldflags=-LIBPATH:C:/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_OPENSSL/install/lib --extra-ldflags=-LIBPATH:C:/Users/garrett/wbpa/git/OpenRV/_build/RV_DEPS_DAV1D/install/lib --enable-openssl --enable-libdav1d
`

**Reply by pablocouto** [reply]
I had what appears to be the same issue. Applying #385 seems to have solved it.

**Reply by wbpa-garrett** [reply]
Oh, thank you! I'll try it out.

**Reply by wbpa-garrett** [reply]
That appears to have fixed it. Thank you! 

Now on to my PySide2 error, but that may be related to the OCF outage and me tinkering with the download to prevent the timeouts? rv_deps_pyside2_SOURCE_DIR is empty for me, but that step got to the install step for me before the outage.

**Reply by pablocouto** [reply]
Glad to know it helped.

About OCF, I edited the URLs in two files, replacing `mirrors.ocf.berkeley.edu/qt/` with `download.qt.io/` in `src/build/make_pyside.py` and `cmake/dependencies/python3.cmake`. This allowed the build to complete.

**Reply by wbpa-garrett** [reply]
That worked great, thanks again! Does anyone know why I'm having an issue opening RV after the install step?

```
Copyright Contributors to the Open RV Project
[19036:6792:0216/141312.500:ERROR:extension_system_qt.cpp(121)] Failed to parse extension manifest.
```

**Comment by krishnaavril** [comment]
How did I manage to rv build 2.1.0? here are my thoughts on getting it working

"pacman -S pip" - don't worry about this

I prefer building this in VM, or you chance it maynot work. Install VScode2022 with the necessary tools shown [here](https://github.com/AcademySoftwareFoundation/OpenRV/blob/main/docs/build_system/win_microsoft_visual_studio_features_to_install.png)

install all necessary packages mentioned in the Windows build, it's not enough(**requests** & some others are missing), to find out  you have to start building, make sure you write out the log and check the errors in the log, if in case its missing any modules install them through pip

latest msys is not supported, you have to get a few steps [down](https://github.com/AcademySoftwareFoundation/OpenRV/issues/549#issue-2494144548)

these things helped me to build successfully!
